interfaces/builtin/personal_files.go

```diff
@@ -0,0 +1,79 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+	"strings"
+)
+
+const personalFilesSummary = `allows access to personal files or directories`
+
+const personalFilesBaseDeclarationPlugs = `
+  personal-files:
+    allow-installation: false
+    deny-auto-connection: true
+`
+
+const personalFilesBaseDeclarationSlots = `
+  personal-files:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const personalFilesConnectedPlugAppArmor = `
+# Description: Can access specific personal files or directories in the 
+# users's home directory.
+# This is restricted because it gives file access to arbitrary locations.
+`
+
+type personalFilesInterface struct {
+	commonFilesInterface
+}
+
+func validateSinglePathHome(np string) error {
+	if !strings.HasPrefix(np, "$HOME/") {
```

Comment:
`$HOME` vs `${HOME}`?

I'm not sure how this looks for consistency with other variable-like syntax we have in interfaces.

Reply:
Not quite sure I fully understand the suggestion but: `${HOME}` does not validate: `"\${HOME}/foo" contains a reserved apparmor char from .*`},` - I added a test for this.